src/App.js

```diff
@@ -14,11 +16,120 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    selectedGood: [' Jam'],
+  }
+
+  clear = () => {
+    this.setState({ selectedGood: [] });
+  }
+
+  toAdd = (good) => {
+    this.setState(prevState => (
+      {
+        selectedGood: [...prevState.selectedGood, ` ${good}`],
+      }
+    ));
+  }
+
+  toRemove = (good) => {
+    this.setState(prevState => (
+      {
+        selectedGood: [...prevState.selectedGood]
+          .filter(el => el !== ` ${good}`),
+      }
+    ));
+  }
+
+  render() {
+    const { selectedGood } = this.state;
+
+    return (
+      <div className="App">
+        <div className="container">
+          <h1 className="header">
+            Goods Selector
+          </h1>
+          <h2 className="selectedList">
+            Selected goods:
+            {
+            selectedGood.length === 1
+              && `${selectedGood} is selected`
+            }
+
+            {
+            selectedGood.length >= 2
+              && `${selectedGood} are selected`
+            }
+
+            {
+            selectedGood.length === 0
+              && ' No goods selected'
+            }
+          </h2>
+
+          {selectedGood.length !== 0
+            && (
+              <>
+                <h2 className="clearText">
+                  Clear selection:
+                </h2>
+
+                <button
+                  type="button"
+                  className="clearButton"
+                  onClick={this.clear}
+                >
+                  X
+                </button>
+              </>
+            )
+          }
+
+          <ul className="list">
+            {goodsFromServer.map(good => (
+              <li
+                key={uuidv4()}
```

Comment:
```suggestion
                key={good}
```
у тебя на каждый рендер будет новый  key, и реакт будет думать что это новый элемент и заново строить Virtual DOM